uniconnect-epfl/uniconnect
# feat: implement google authentication

# What I did
 I implemented the login functionality with google. I have also implemented tests for login with firebase and with google

# How I did it

- I added onto the actual google login screen, the logic behind the call to google API
- To make the login work I installed a new `@react-native-google-signin/google-signin` library and connected it with firebase.

# How to test it

When running the code, you can print the "user" const in OnboardingScreen.tsx and see that it returns a token as well as the other user values from google.

Testing with Expo Go does not work for this feature, thus you should start your own local simulator (with Expo Orbit for example) and download an APK. [This APK](https://expo.dev/accounts/uniconnect-epfl/projects/uniconnect/builds/88c246fc-aa61-433a-b803-749246ab8c91
) is already a valid one to test the functionality 

# What has been unit tested?

I have tested the Onboarding Screen functionalities. I have mocked the `loginEmailPassword` and the `GoogleSignin.signIn()` functions, which means that the backend hasn't been tested. This is done on purpose so that backend and frontend testing is separated. 

Thus, an upcoming PR should be to test the backend functionalities thoroughly. 

# Demo video

https://github.com/uniconnect-epfl/uniconnect/assets/91184289/c37c75a5-5e17-4687-a852-5275f6c9e1d1


# Pre-merge checklist
The changes I have introduced:
